wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUXRefClient.java

```diff
@@ -0,0 +1,557 @@
+    }
+
+    /**
+     * Initializes the service's underlying stub Should only be used by constructors.
+     *
+     * @param conn
+     *            -- All connection settings included
+     */
+    protected void initWsDFUXRefClientStub(Connection conn)
+    {
+        try
+        {
+            setActiveConnectionInfo(conn);
+            stub = setStubOptions(new WsDFUXRefStub(conn.getBaseUrl() + WSDFUXREFURI), conn);
+        }
+        catch (AxisFault e)
+        {
+            log.error("Could not initialize WsDFUXRefStub - Review all HPCC connection values");
+            e.printStackTrace();
+        }
+        catch (Exception e)
+        {
+            log.error("Could not initialize WsDFUXRefStub - Review all HPCC connection values");
+            if (!e.getLocalizedMessage().isEmpty())
+            {
+                initErrMessage = e.getLocalizedMessage();
+                log.error(e.getLocalizedMessage());
+            }
+        }
+    }
+
+    /**
+     * @param processCluster
+     * @param getFileDetails
+     * @param checkPackageMaps
+     * @return
+     * @throws Exception
+     */
+    public DFUXRefUnusedFilesResponseWrapper unusedFiles(String processCluster, Boolean getFileDetails, Boolean checkPackageMaps) throws Exception
+    {
+        verifyStub();
+
+        DFUXRefUnusedFilesRequest request = new DFUXRefUnusedFilesRequest();
+        request.setProcessCluster(processCluster);
+        if (checkPackageMaps != null)
+            request.setCheckPackageMaps(checkPackageMaps);
+        if (getFileDetails != null)
+            request.setGetFileDetails(getFileDetails);
+
+        DFUXRefUnusedFilesResponse resp = null;
+
+        try
+        {
+            resp = ((WsDFUXRefStub) stub).dFUXRefUnusedFiles(request);
+        }
+        catch (RemoteException e)
+        {
+            throw new Exception("HPCCWsDFUXRefClient.unusedFiles(" + processCluster + ") encountered RemoteException.", e);
+        }
+
+        if (resp.getExceptions() != null)
+        {
+            ArrayOfEspException exceptions = resp.getExceptions();
+            handleEspExceptions(new ArrayOfEspExceptionWrapper(exceptions), "Error fetching unusedFiles: '" + processCluster + "'");
+        }
+
+        return new DFUXRefUnusedFilesResponseWrapper(resp);
+    }
+    /**
+     * @param cluster
+     * @return
+     * @throws Exception
+     */
+    public String orphanedFiles(String cluster) throws Exception
+    {
+        verifyStub();
+
+        DFUXRefOrphanFilesQueryRequest request = new DFUXRefOrphanFilesQueryRequest();
+        request.setCluster(cluster);
+
+        DFUXRefOrphanFilesQueryResponse resp = null;
+
+        try
+        {
+            resp = ((WsDFUXRefStub) stub).dFUXRefOrphanFiles(request);
+        }
+        catch (RemoteException e)
+        {
+            throw new Exception("HPCCWsDFUXRefClient.orphanedFiles(" + cluster + ") encountered RemoteException.", e);
+        }
```

Comment:
Do we need to check the resp for exceptions here as well?

Reply:
Good eye- the implementation of the methods is inconsistent, some declare inline exceptions, some don't...